src/FileSystem/FileSystem.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace OomphInc\WASP\FileSystem;
+
+use RuntimeException;
+
+class FileSystem implements FileSystemInterface {
+
+	use DirStackTrait;
+
+	/**
+	 * @param string [$cwd] optional starting current working directory
+	 */
+	public function __construct($cwd = null) {
+		$this->setCwd($cwd);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function resolvePath($path) {
+		// default path if null provided
+		if ($path === null) {
+			return getcwd();
+		}
+
+		// is the path relative?
+		if (!preg_match(sprintf('#^%1$s|[a-z]+:%1$s{1,2}#i', preg_quote(DIRECTORY_SEPARATOR)), $path)) {
+			$path = $this->getCwd() . '/' . $path;
```

Comment:
another hardcoded '/' since you're actually doing it!

Reply:
gah!